crates/uv-toolchain/src/discovery.rs

```diff
@@ -1237,6 +1237,9 @@ impl FromStr for VersionRequest {
             Ok(selector)
         // e.g. `>=3.12.1,<3.12`
         } else if let Ok(specifiers) = VersionSpecifiers::from_str(s) {
```

Comment:
What is `specifiers` in this case?

Reply:
Sorry can you rephrase? It's an empty `VersionSpecifiers` that allows any version.

Reply:
How does `"python"` get parsed as an empty `VersionSpecifiers`? I'm just trying to understand the data flow.

Reply:
Ah yes I can answer that :)

https://github.com/astral-sh/uv/blob/5f37395f4537026529c8dcb38a9151c9d45d388a/crates/uv-toolchain/src/discovery.rs#L883-L888

Previously an empty remainder here would result in us not treating this as a Python version request, now it does. I didn't expect `VersionSpecifiers` to allow empty strings, but it makes sense in hindsight.

I'm going to need to split `VersionRequest::Range` out of `VersionRequest` (or something like that) to have the user experience I want — trying to figure out how best to do that next.